c2rust-analyze/src/rewrite/expr/distribute.rs

```diff
@@ -30,6 +30,15 @@ struct RewriteInfo {
     loc: Location,
     sub_loc: Vec<SubLoc>,
     desc: MirOriginDesc,
+    priority: Priority,
+}
+
+/// This enum defines a sort order for [`RewriteInfo`], from innermost (applied earlier) to
+/// outermost (applied later).
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum Priority {
+    Eval,
+    LoadResult,
```

Comment:
Docs for these would be nice, as I'm not exactly what the names are supposed to mean.

Reply:
Added a comment explaining the ordering: a2ebd4f1